src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -117,5 +124,31 @@ protected override void EnrichTelemetryApiEvent(ApiEvent apiEvent)
 
             return dict;
         }
+
+        private async Task<MsalTokenResponse> CheckForBrokerAndSendTokenRequestAsync(MsalRefreshTokenCacheItem msalRefreshTokenItem, CancellationToken cancellationToken)
```

Comment:
>CheckForBrokerAndSendTokenRequestAsync [](start = 46, length = 38)

duplicate code ? why?

Reply:
now that i can save to keychain, i will dive more into the silent flow. we need to send a new parameter and we won't have the ownerUiParent, so there are a few small differences, but not many. i'll take a look at this.

---
In reply to: [256614448](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r256614448) [](ancestors = 256614448)

Reply:
Please organize this code enabling it the differences to be passed in but all the similarities should be in one place.

---
In reply to: [257441318](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r257441318) [](ancestors = 257441318,256614448)